text/trace/0170-sampling-probability.md

```diff
@@ -0,0 +1,705 @@
+Context and whether it was sampled, in order to calculate the
+probability of starting to sample a new "sub-root" in the trace.
+
+Using standard notation for conditional probability, `P(x)` indicates
+the probability of `x` being true, and `P(x|y)` indicates the
+probability of `x` being true given that `y` is true.  The axioms of
+probability establish that:
+
+```
+P(x)=P(x|y)*P(y)+P(x|not y)*P(not y)
+```
+
+The variables are:
+
+- **`H`**: The head inclusion probability of the parent context that
+  is in effect, independent of whether the parent context was sampled
+- **`I`**: The inflationary sampling probability for the span being
+  started.
+- **`D`**: The decision probability for whether to start a new sub-root.
+
+This Sampler cannot lower sampling probability, so if the new span is
+started with `H >= I` or when the context is already sampled, no new
+sampling decisions are made.  If the incoming context is already
+sampled, the adjusted count of the new span is `1/H`.
+
+Assuming `H < I` and the incoming context was not sampled, we have the
+following probability equations:
+
+```
+P(span sampled) = I
+P(parent sampled) = H
+P(span sampled | parent sampled) = 1
+P(span sampled | parent not sampled) = D
+```
+
+Using the formula above, 
+
+```
+I = 1*H + D*(1-H)
+```
+
+solve for D:
+
+```
+D = (I - H) / (1 - H)
+```
+
+Now the Sampler makes a decision with probability `D`.  Whether the
+decision is true or false, propagate `I` as the new head inclusion
+probability.  If the decision is true, begin recording a sub-rooted
+trace with adjusted count `1/I`.
+
+## Proposed `Span` protocol
+
+Earlier drafts of this document had proposed the use of Span
+attributes to convey a the combined effects of head- and tail-sampling
+in the form of an (optional) adjusted count and (optional) sampler
+name.  The group did not reach agreement on whether and/or how to
+convey tail sampling.
+
+Following the proposal for propagating consistent head trace sampling
+probability developed in [OTEP
+168](https://github.com/open-telemetry/oteps/pull/168), this proposal
+is limited to adding a field to encode the head sampling probability.
+The OTEP 168 proposal for propagation limits head sampling
+probabilities to powers of two, hence we are able to encode the
+corresponding adjusted count using a small non-negative integer.
+
+Interoperability with existing Propagators and Span data means
+recognizing Spans with unknown adjusted count when the new field is
+unset.  Thus, the 0 value shall mean unknown adjusted count.
+
+The OTEP 168 proposal for _propagating_ head sampling probability uses
+6 bits of information, with 62 ordinary values, one zero value, and a
+single unused value.
+
+Here, we propose a biased encoding for head sampling probability equal
+to 1 plus the `P` value as proposed in OTEP 168.  The proposed span
+field, a biased base-2 logarithm of the adjusted count, is named
+simply `log_head_adjusted_count` and still requires 6 bits of
```

Comment:
I am wondering, if `log_head_adjusted_count` should be rather named `log_parent_adjusted_count`. Consider, a trace A -> B -> C where A and B are consistently sampled using different sampling rates and C uses a parent sampler that samples the span if the parent B is sampled. I guess we want to know the sampling rate of B for span C. Therefore, the parent sampling rate needs to be propagated and not that of the head.

Reply:
I think the intention is to describe *current* span and abstract away the origin of adjusted count calculation.

Reply:
We discussed this and agreed to take this question to spec PR #1899. This OTEP can merge either way.
Terms that we consider alternatives to "head":
"parent" (as in this comment)
"initial" (another option)

Reply:
How about simply `log_adjusted_count`?